packages/mui-codemod/src/v6.0.0/list-item-button-prop/list-item-button-prop.js

```diff
@@ -0,0 +1,58 @@
+import findComponentJSX from '../../util/findComponentJSX';
+
+/**
+ * @param {import('jscodeshift').FileInfo} file
+ * @param {import('jscodeshift').API} api
+ */
+export default function transformer(file, api, options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  const printOptions = options.printOptions;
+
+  //Rename components that have ListItem button to ListItemButton
+  findComponentJSX(j, { root, componentName: 'ListItem' }, (elementPath) => {
+    const index = elementPath.node.openingElement.attributes.findIndex(
+      (attr) => attr.type === 'JSXAttribute' && attr.name.name === 'button',
+    );
+    if (index !== -1) {
+      elementPath.node.openingElement.name.name = 'ListItemButton';
+      elementPath.node.openingElement.attributes.splice(index, 1);
+    }
+  });
+
+  //Find if there are ListItem imports/named imports.
+  let containsListItemImport = root.find(j.ImportDeclaration).filter(path => path.node.source.value === '@mui/material');
+  let containsListItemNamedImport = containsListItemImport.find(j.ImportSpecifier).filter(path => path.node.imported.name === 'ListItem');
+  let containsListItem = false;
+
+  //Find components that use ListItem. If they do, we shouldn't remove it
+  findComponentJSX(j, { root, componentName: 'ListItem' }, (elementPath) => {
+    if(elementPath.node.openingElement.name.name === 'ListItem') {
+      containsListItem = true;
+    }
+  });
+
+  //Remove ListItem import if there is no usage
+  if(containsListItemNamedImport.length === 0 || !containsListItem) {
+    // root.find(j.ImportDeclaration).filter(path => path.node.source.value === '@mui/material').find(j.ImportSpecifier)
+    // .filter(path => path.node.imported.name === 'ListItem').remove();
+    root.find(j.ImportDeclaration).filter(path => path.node.source.value === '@mui/material/ListItem').remove();
+  }
```

Comment:
Let's try this (I haven't tried it so it might need some fixing):

```suggestion
  //Find components that use ListItem. If they do, we shouldn't remove it
  findComponentJSX(j, { root, componentName: 'ListItem' }, (elementPath) => {
    if(elementPath.node.openingElement.name.name === 'ListItem') {
      containsListItem = true;
    }
  });
 
  //Find if there are ListItem imports/named imports.
  let containsListItemImport = root.find(j.ImportDeclaration).filter(path => path.node.source.value === '@mui/material');
  let containsListItemNamedImport = containsListItemImport.find(j.ImportSpecifier).filter(path => path.node.imported.name === 'ListItem');
  let containsListItem = false;

  // Remove ListItem imports if there is no usage
  if(!containsListItem) {
    // Remove named imports
    root.find(j.ImportDeclaration).filter(path => path.node.source.value === '@mui/material').find(j.ImportSpecifier).filter(path => path.node.imported.name === 'ListItem').remove();
 
    // Remove default imports
    root.find(j.ImportDeclaration).filter(path => path.node.source.value === '@mui/material/ListItem').remove();
  }
```